index.js

```diff
@@ -67,11 +69,12 @@ function traceModuleFunction(module, emit) {
 
         const event = new Event(name);
         for (let i = 0; i !== numInputActions; i++) {
-          const item = inputActions[i];
-          const action = item[0];
-          const params = item[1];
+          const [ action, params ] = inputActions[i];
           action(values, event, params);
         }
+        if (isFunction(spec.callbacks.onEnter)) {
+          spec.callbacks.onEnter(event, this.context);
```

Comment:
The idea was to avoid exposing all the capabilities and try to make the api declarative, in theory the event should contain the arguments parsed already. and context is the only thing missing from the event. Maybe @oleavr can share his feelings on this too